src/game_api/script/usertypes/entities_items_lua.cpp

```diff
@@ -819,6 +820,21 @@ void register_usertypes(sol::state& lua)
         sol::base_classes,
         sol::bases<Entity, Movable>());
 
+    lua.new_usertype<OlmecShip>(
+        "OlmecShip",
+        "sound",
+        &OlmecShip::sound,
+        "door_fx",
+        &OlmecShip::door_fx,
+        "smoke",
+        &OlmecShip::smoke,
+        "flight_time",
+        &OlmecShip::flight_time,
+        "has_spawned_jetflames",
+        &OlmecShip::has_spawned_jetflames,
+        sol::base_classes,
+        sol::bases<Entity, Movable>());
+
```

Comment:
The proper way of adding new entity types is: add them in plugin: run `entity_casting.py` (it will generate new `entities-hierarchy.md`) then copy the output of the console to `entity_casting_lua.cpp` and `lua_console.cpp`

Reply:
Ok, I think I may have done it right.